### DISCUSSION TITLE
【功能建议】mac panel图标可以右键弹出菜单选择哪一个实例

### DISCUSSION THREAD

**Discussion by widon1104** [linuxdeepin/developer-center#2977] [Features Request & Ideas | 特性请求 & 头脑风暴]
![WechatIMG4](https://user-images.githubusercontent.com/12093211/105272703-dcc4a000-5bd4-11eb-89ac-53a0638c94c7.jpeg)
mac panel图标可以右键弹出菜单选择哪一个实例，我觉得比现在deepin（UOS）面板的处理要好，希望deepin能模仿一下

**Comment by justforlxz** [comment]
那是因为mac的panel没有窗口预览，我们的选择是通过预览画面进行切换的。

**Comment by widon1104** [comment]
我在家使用deepin，公司使用mac，窗口预览本来就很模糊，还是要看实例名是啥，我觉得使用窗口预览切换实例还没有使用列表切换来的快

**Comment by justforlxz** [comment]
> 我在家使用deepin，公司使用mac，窗口预览本来就很模糊，还是要看实例名是啥，我觉得使用窗口预览切换实例还没有使用列表切换来的快

考虑了一下，似乎只需要在dde-dock上修改就可以了，如果你想参与开发，我可以指导你完成这个功能（

**Comment by widon1104** [comment]
使用列表切换，只要在图标上面右键，所有实例名字净收眼底，可以快速选择。而使用预览画面，需要在一个画面上面停留1秒钟才知道单个实例名称是啥，我感觉比列表切换慢很多。

我不懂qt开发，不能参与，这个功能其实看起来不难，只用加一个右键弹出列表，点击切换就可以了，预览画面都可以保留。

**Comment by justforlxz** [comment]
> 使用列表切换，只要在图标上面右键，所有实例名字净收眼底，可以快速选择。而使用预览画面，需要在一个画面上面停留1秒钟才知道单个实例名称是啥，我感觉比列表切换慢很多。
> 
> 我不懂qt开发，不能参与，这个功能其实看起来不难，只用加一个右键弹出列表，点击切换就可以了，预览画面都可以保留。

很简单的，我可以指导你在什么地方提交什么样的代码（

https://github.com/linuxdeepin/dde-dock/blob/master/frame/item/dockitem.cpp#L230 这个函数负责显示看到的那个菜单，只需要在这里获取一下窗口列表，并往菜单里添加新的内容就可以了。然后等选择以后在 https://github.com/linuxdeepin/dde-dock/blob/master/frame/item/dockitem.cpp#L263 进行一个判断，判断出是你添加的窗口列表的action,就可以调用后端的接口激活对应窗口，就完成切换了（很简单吧

**Comment by justforlxz** [comment]
![2021-01-23 21-18-07屏幕截图](https://user-images.githubusercontent.com/12298476/105579707-0ba06900-5dc3-11eb-83e6-6ee04464fa82.png)
![2021-01-23 21-18-43屏幕截图](https://user-images.githubusercontent.com/12298476/105579709-0e9b5980-5dc3-11eb-8901-210f723319b2.png)
![2021-01-23 21-19-25屏幕截图](https://user-images.githubusercontent.com/12298476/105579710-0f33f000-5dc3-11eb-915b-c2668aaebb9e.png)

cannot upload gif, it's so big.

**Comment by justforlxz** [comment]
https://youtu.be/9JqckIPZHD0

**Comment by justforlxz** [comment]
老哥，你不能每次都是上来提几个需求就不回复了啊

**Comment by justforlxz** [comment]
老哥，你不能每次都是上来提几个需求就不回复了啊

**Comment by widon1104** [comment]
跟mac做的一样,然后通过内部测试就可以了啊,还回复啥啊

**Comment by justforlxz** [comment]
> 跟mac做的一样,然后通过内部测试就可以了啊,还回复啥啊

这是你提的需求，你起码对我做的结果给个意见啊，你要是只提出需求，不关心过程和结果，那我们还怎么开展后续的工作…

如果你完全不care我们到底要不要实现你的需求，那以后你提的需求怕是没人愿意做了。

**Comment by widon1104** [comment]
我在这里面提的95%以上的需求,都没有人做啊,有时候都不抱啥希望了.
我前几天看了一下,看起来还可以,需要特别指出来的一点,在图标上面右键只是弹出实例菜单,还是需要点击左键来选择,视频里面看不出你鼠标怎么点击的.

**Comment by justforlxz** [comment]
> 我在这里面提的95%以上的需求,都没有人做啊,有时候都不抱啥希望了.

说实话，你提的大部分需求，我个人不怎么认同的，好功能太多了，但是没办法全部缝合到一个系统身上。
这两年公司忙于生计，忽略了社区，现在安排的新的团队独立管理和运营社区，最近我们在整理所有的需求，等待基本的评估以后，会给出做不做的回应。


> 我前几天看了一下,看起来还可以,需要特别指出来的一点,在图标上面右键只是弹出实例菜单,还是需要点击左键来选择,视频里面
看不出你鼠标怎么点击的.

可以是可以，但是我询问了几个社区的用户，他们都觉得在这里加列表不如优化一下窗口预览，例如加上一个标题栏之类的。

**Comment by widon1104** [comment]
窗口预览根本不行,没列表方便,而且加上这个功能以后可以保留窗口预览吧

**Comment by BLumia** [comment]
@widon1104 

> 我在这里面提的95%以上的需求,都没有人做啊,有时候都不抱啥希望了.

做下简单说明：deepin 是社区产品，UOS 是专业版产品。前后两者无论是 bug 反馈还是需求反馈的处理流程都是不一样的。UOS 产品通过它对应的渠道反馈后会有人和您对接问题协商处理，而 deepin 作为社区产品，如果我们这边未能通过产品规划来采纳纳入开发的功能，就只能期待社区来贡献了。并不是说来自社区提的任何需求我们都会无条件做，当无法纳入时，是否做、怎么做当然就取决于社区参与人员要讨论的东西了。

您这个需求我这边标记下看看产品那边能否纳入到产品功能中。
